Add new rule `a11y-no-title-usage` (#316)
primer/eslint-plugin-primer-react: docs/rules/a11y-no-title-usage.md

## Rule Details

This rule aims to prevent the use of the `title` attribute with some components from `@primer/react`. The `title` attribute is not keyboard accessible, which results in accessibility issues. Instead, we should utilize alternatives that are accessible.

👎 Examples of **incorrect** code for this rule

```jsx
import {RelativeTime} from '@primer/react'

const App = () => <RelativeTime date={new Date('2020-01-01T00:00:00Z')} noTitle={false} />
```

👍 Examples of **correct** code for this rule:

```jsx
import {RelativeTime} from '@primer/react'

const App = () => <RelativeTime date={new Date('2020-01-01T00:00:00Z')} />
```

> The `noTitle` attribute can be omitted because its default value is `true` internally.

Reviewer: Maybe "The `noTitle` attribute is `true` by default, so it can be omitted." is a bit easier to read? Just nit picking here

Author: I like this, your suggestion reads better! Thank you 😄

> The `noTitle` attribute can be omitted because its default value is `true` internally.

Author: ```suggestion
The noTitle attribute is true by default, so it can be omitted.
```

> ## With alternative tooltip

Reviewer: Great to have this example ❤️
